From 05dafd2f25a41f6b0b66b5da0b9c2778b0637003 Mon Sep 17 00:00:00 2001
From: christos <christos>
Date: Sun, 10 May 2015 13:28:54 +0000
Subject: [PATCH 1/1] handle eval sleep 10^C not exiting, by setting doneinp if
 we are interrupted inside an eval from paulo.cesar.pereira.de.andrade

---
 sh.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sh.c b/sh.c
index 14e8433..c0dd8bc 100644
--- a/sh.c
+++ b/sh.c
@@ -1942,6 +1942,8 @@ pintr1(int wantnl)
     {
 	(void) Cookedmode();
 	GettingInput = 0;
+	if (evalvec)
+	    doneinp = 1;
     }
     drainoline();
 #ifdef HAVE_GETPWENT
-- 
2.5.5

